### PR TITLE
Fix regenerate exercise M4.01 from solution

### PR DESCRIPTION
**Pull request by ArturoAmorQ**
PR #422 erased a hint in [exercise M4.01](https://inria.github.io/scikit-learn-mooc/python_scripts/linear_models_ex_01.html).
This fixes it while improving wording.

### CODE REVIEW

File: python_scripts/linear_models_sol_01.py [hunk 1]
```diff
@@ -46,9 +46,12 @@ def linear_model_flipper_mass(
...
 #
-# Given a vector of the flipper length, several weights and intercepts to
-# plot several linear model that could fit our data. Use the above
-# helper function to visualize both the model and data.
+# Define a vector ```weights = [...]``` and a vector ```intercepts = [...]``` of
+# the same length. Each pair of entries ```(weights[i], intercepts[i])``` tags a
+# different model. Use these vectors along with the vector
+# ```flipper_length_range``` to plot several linear models that could possibly
+# fit our data. Use the above helper function to visualize both the models and
+# the real samples.
```
Comment (glemaitre) [inline python_scripts/linear_models_sol_01.py:54]: is it normal to have three backticks to define a text as code here?

File: python_scripts/linear_models_sol_01.py [hunk 2]
```diff
@@ -93,10 +96,10 @@ def goodness_fit_measure(true_values, predictions):
     # we compute the error between the true values and the predictions of our
     # model
     errors = np.ravel(true_values) - np.ravel(predictions)
-    # We have several possible strategy to reduce all errors to a single value.
-    # Computing the mean error (sum divided by the number of element) looks
-    # like a good solution. However, we have negative error and therefore, we
-    # need to sum only positive numbers. Therefore, we can either square each
+    # We have several possible strategies to reduce all errors to a single value.
+    # Computing the mean error (sum divided by the number of element) could seem
+    # like a good solution. However, we have negative errors that will misleadingly
```
Comment (glemaitre) [inline python_scripts/linear_models_sol_01.py:101]: "could seem like" seems odd to me :)

Comment (ArturoAmorQ) [reply python_scripts/linear_models_sol_01.py:101]: We can alternatively use "[might seem like](https://www.phrasemix.com/phrases/it-might-seem-like-something-but-sentence)", which is a little better. What do you think?

File: python_scripts/linear_models_sol_01.py (hunk 1 above)
Comment (ArturoAmorQ) [reply python_scripts/linear_models_sol_01.py:54]: I had in mind that three back-ticks are good for quoting code in a particular format, though this is probably unnecessary here.

Comment (ogrisel) [reply python_scripts/linear_models_sol_01.py:54]: Triple backticks are probably only useful for multi-line code blocks.

File: python_scripts/linear_models_sol_01.py (hunk 2 above)
Comment (ogrisel) [reply python_scripts/linear_models_sol_01.py:101]: +1 for "might seem like" or just "seems like" as you prefer.

**Comment by glemaitre** [comment]
Thanks @ArturoAmorQ

**Comment by lesteve** [comment]
Note to ourselves: each time we change a solution we need to regenerate the corresponding exercise.

**Comment by lesteve** [comment]
I just pushed a commit updating the exercise

**Pull request merged**